src/mafiasi_link_shortener/api/views.py

```diff
@@ -16,11 +20,18 @@
 class LinkViewset(viewsets.ModelViewSet):
     serializer_class = serializers.LinkSerializer
     queryset = models.Link.objects.all().select_related("owner").order_by("short")
-    permission_classes = [HasTokenScope & IsAuthenticated & IsOwner]
+    permission_classes = [(HasTokenScope | HasSessionScope) & IsAuthenticated & IsOwner]
     lookup_field = "short"
 
     def get_queryset(self):
         # only filter when listing links because we deny access in other cases
         if self.action == "list":
             return super().get_queryset().filter(owner__exact=self.request.user)
         return super().get_queryset()
+
+
+def logged_in(request):
+    if request.user.is_authenticated:
+        return HttpResponse(status=HTTPStatus.OK)
+    else:
+        return HttpResponse(status=HTTPStatus.UNAUTHORIZED)
```

Comment:
Does this work properly with schema generation?

Reply:
Fixed in 1bf12867caa35d89875120fdb327ce259e33469a.